presto-main/src/main/java/com/facebook/presto/execution/executor/TaskExecutor.java

```diff
@@ -74,19 +75,30 @@
 import static com.google.common.base.Preconditions.checkArgument;
 import static com.google.common.base.Preconditions.checkState;
 import static com.google.common.collect.Sets.newConcurrentHashSet;
+import static java.lang.System.lineSeparator;
+import static java.util.Arrays.asList;
 import static java.util.Objects.requireNonNull;
 import static java.util.concurrent.Executors.newCachedThreadPool;
 import static java.util.concurrent.Executors.newSingleThreadScheduledExecutor;
 import static java.util.concurrent.TimeUnit.MICROSECONDS;
 import static java.util.concurrent.TimeUnit.NANOSECONDS;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static java.util.stream.Collectors.joining;
+
 
 @ThreadSafe
 public class TaskExecutor
 {
     private static final Logger log = Logger.get(TaskExecutor.class);
 
     // print out split call stack if it has been running for a certain amount of time
-    private static final Duration LONG_SPLIT_WARNING_THRESHOLD = new Duration(600, TimeUnit.SECONDS);
+    private static final Duration LONG_SPLIT_WARNING_THRESHOLD = new Duration(600, SECONDS);
+    // Interrupt a split if it is running longer than this AND it's blocked on something known
+    private static final Duration LONG_SPLIT_INTERRUPT_THRESHOLD = new Duration(600, SECONDS);
+    private static final Predicate<List<StackTraceElement>> INTERRUPTIBLE_SPLIT_PREDICATE = elements ->
+                    elements.stream()
+                            .anyMatch(element -> element.getClassName().equals(JoniRegexpFunctions.class.getName()));
+    private static final long DEFAULT_INTERRUPT_SPLIT_INTERVAL = 60;
```

Comment:
`DEFAULT_INTERRUPT_SPLIT_INTERVAL_IN_SECONDS`

Reply:
Renamed, changed to a duration